src/broker/broker.c

```diff
@@ -1965,13 +1967,75 @@ static void module_status_cb (module_t *p, int prev_status, void *arg)
     }
 }
 
+static bool signal_is_deadly (int signum)
+{
+    int deadly_sigs[] = { SIGHUP, SIGINT, SIGQUIT, SIGTERM, SIGALRM };
+    for (int i = 0; i < ARRAY_SIZE (deadly_sigs); i++) {
+        if (signum == deadly_sigs[i])
+            return true;
+    }
+    return false;
+}
+
+static void killall_cb (flux_future_t *f, void *arg)
+{
+    broker_ctx_t *ctx = arg;
+    int count = 0;
+    if (flux_rpc_get_unpack (f, "{s:i}", "count", &count) < 0) {
+        flux_log_error (ctx->h,
+                        "job-manager.killall: %s",
+                        future_strerror (f, errno));
+    }
+    flux_future_destroy (f);
+    if (count) {
+        flux_log (ctx->h,
+                  LOG_INFO,
+                  "forwarded signal %d to %d jobs",
+                  (int) ptr2int (flux_future_aux_get (f, "signal")),
+                  count);
+    }
+}
+
+static int killall_jobs (broker_ctx_t *ctx, int signum)
+{
+    flux_future_t *f = NULL;
+    if (!(f = flux_rpc_pack (ctx->h,
+                    "job-manager.killall",
+                    FLUX_NODEID_ANY,
+                    0,
+                    "{s:b s:i s:i}",
+                    "dry_run", 0,
+                    "userid", FLUX_USERID_UNKNOWN,
+                    "signum", signum))
+        || flux_future_then (f, -1., killall_cb, ctx) < 0) {
+            flux_future_destroy (f);
+            return -1;
+    }
```

Comment:
We usually indent the body of blocks like this by one tab so it begins where the `||` is.

Reply:
Yeah, I know, good catch. When these lines were moved to their own function the formatting must have been preserved.

Reply:
Actually, don't we line up arguments in indented functions, even in a conditional? e.g.

https://github.com/flux-framework/flux-core/blob/5948b01cd86592f6beef89624fa8d6f7654a7a4c/src/cmd/builtin/filemap.c#L222-L228

Note the `||` is indented by one tab. Still the current indentation is wrong